src/components/Tool/MergeContacts/MergeContacts.tsx

```diff
@@ -91,40 +100,112 @@ const MergeContacts: React.FC<Props> = ({
     }
   };
 
-  const mergeContacts = async () => {
-    const mergeActions = Object.entries(actions).filter(
-      (action) => action[1].action === 'merge',
-    );
-    if (mergeActions.length) {
-      const winnersAndLosers: { winnerId: string; loserId: string }[] =
-        mergeActions.map((action) => {
-          return { winnerId: action[0], loserId: action[1].mergeId || '' };
-        });
-      await contactsMerge({
-        variables: {
-          input: {
-            winnersAndLosers,
-          },
-        },
-        update: (cache) => {
-          // Delete the loser contacts and remove dangling references to them
-          winnersAndLosers.forEach((contact) => {
-            cache.evict({ id: `Contact:${contact.loserId}` });
+  const handleBulkUpdateDuplicates = async () => {
+    try {
+      const callsByDuplicate: (() => Promise<{ success: boolean }>)[] = [];
+
+      const mergeActions = Object.entries(actions).filter(
+        (action) => action[1].action === 'merge',
+      );
+
+      if (mergeActions.length) {
+        const winnersAndLosers: { winnerId: string; loserId: string }[] =
+          mergeActions.map((action) => {
+            return { winnerId: action[0], loserId: action[1].mergeId || '' };
           });
-          cache.gc();
-        },
-        onCompleted: () => {
-          enqueueSnackbar(t('Success!'), {
+
+        const callMergeDuplicatesMutation = () =>
+          mergeDuplicates(winnersAndLosers);
+        callsByDuplicate.push(callMergeDuplicatesMutation);
+      }
+
+      const duplicatesToIgnore = Object.entries(actions)
+        .filter((action) => action[1].action === 'ignore')
+        .map((action) => action[0]);
+
+      if (duplicatesToIgnore.length) {
+        duplicatesToIgnore.forEach((duplicateId) => {
+          const callIgnoreDuplicateMutation = () =>
+            ignoreDuplicates(duplicateId);
+          callsByDuplicate.push(callIgnoreDuplicateMutation);
+        });
+      }
+
+      if (callsByDuplicate.length) {
+        const results = await Promise.all(
+          callsByDuplicate.map((call) => call()),
+        );
+
+        const failedUpdates = results.filter(
+          (result) => !result.success,
+        ).length;
+        const successfulUpdates = results.length - failedUpdates;
+
+        if (successfulUpdates) {
+          enqueueSnackbar(t(`Updated ${successfulUpdates} duplicate(s)`), {
             variant: 'success',
           });
+        }
+        if (failedUpdates) {
+          enqueueSnackbar(
+            t(`Error when updating ${failedUpdates} duplicate(s)`),
+            {
+              variant: 'error',
+            },
+          );
```

Comment:
I think it would be nice if it alerted to the user which ones had an error

Reply:
Right now the API only returns the winner id's of the merges that were successful. It would be a lot of work to find the names of those who didn't. Instead, the ones that failed will stay in the list of duplicates for the user to try again.